misc/compiler-plugins/modules/configurable-schema-generator/src/main/java/org/ballerinalang/configschemagenerator/ConfigSchemaGenTask.java

```diff
@@ -0,0 +1,107 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.configschemagenerator;
+
+import io.ballerina.projects.ConfigReader;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.ProjectException;
+import io.ballerina.projects.ProjectKind;
+import io.ballerina.projects.plugins.AnalysisTask;
+import io.ballerina.projects.plugins.CompilationAnalysisContext;
+import io.ballerina.projects.util.ProjectConstants;
+import org.ballerinalang.configschemagenerator.util.ConfigSchemaBuilder;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+
+/**
+ * Task to generate JSON schema for configurable variables.
+ *
+ * @since 2.0.0
+ */
+public class ConfigSchemaGenTask implements AnalysisTask<CompilationAnalysisContext> {
+
+    static final String CONFIG_SCHEMA = "config-schema.json";
+
+    @Override
+    public void perform(CompilationAnalysisContext compilationAnalysisContext) {
+        String schema = ConfigSchemaBuilder.getConfigSchemaContent(ConfigReader.getConfigVariables(
+                compilationAnalysisContext.currentPackage()));
+        writeConfigJSONSchema(schema, compilationAnalysisContext.currentPackage().project());
+    }
+
+    /**
+     * Write config schema JSON to the target/bin path.
+     *
+     * @param schema JSON schema content as String
+     * @param project Project instance
+     */
+    private void writeConfigJSONSchema(String schema, Project project) {
+        Path path;
+        if (project.kind().equals(ProjectKind.SINGLE_FILE_PROJECT)) {
+            path = project.sourceRoot().getParent();
```

Comment:
Should we use the current directory here?

Reply:
`project.sourceRoot()` returns the path including the `bal` file. So, had to use `getParent()` to retrieve the current directory.

Reply:
Fixed with 0f8c1a2